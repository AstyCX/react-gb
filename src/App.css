* {
  padding: 0;
  margin: 0;
}

body {
  background: aliceblue;
}

.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  height: 25vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-main {
  display: flex;
  justify-content: space-between;
  min-height: 75vh;
}

.message {
  margin-bottom: 15px;
  height: content-box;
}

.messageText, .messageAuthor {
  color: black;
  font-size: 15px;
  font-weight: bold;
}

.messages {
  padding: 10px 0;
  box-sizing: border-box;
  display: flex;
  justify-content: space-around;
  width: 100%;
  flex-wrap: wrap;
}

.carryMessages {
  display:flex;
  flex-direction: column;
  width: 100%;
}

.sendMess {
  display: flex;
  align-items: center;
  justify-content: center;
}

.messageArea {
  width: 100%;
}

.nameArea {
  width: 100%;
}

.active {
  color: #5151c1;
}

.changeName {
  padding: 20px;
  max-height: 30px;
  display: flex;
  align-items: center;
  width: 100%;
}

.gists {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.gistsDescription {
  margin: 0 0 15px 0;
}

.formRegistration, .formLogin {
  width: 100%;
  max-width: 520px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  max-height: 190px;
  padding: 10px;
  box-sizing: border-box;
  height: 100%;
}

.errorFormRegistration, .errorFormLogin {
  max-height: 230px;
}

.registration, .login {
  width: 100%;
}

.loading {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

