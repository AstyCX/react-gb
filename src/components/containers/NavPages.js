import {Tabs, Tab} from "@mui/material";
import {Link, useLocation} from 'react-router-dom';
import {useEffect, useState} from "react";

const NavPages = ({pages}) => {
    const location = useLocation();
    const [loc, setLoc] = useState(location.pathname.split(/\//)[1]);

    useEffect(()=>{
        setLoc(location.pathname.split(/\//)[1])
    }, [location])

    return (
        <Tabs value={loc} aria-label="nav tabs example">
            {Object.keys(pages).map((el,i)=>(
                <Tab value={pages[el].id} key={i} label={el} to={pages[el].link} component={Link}/>))}
        </Tabs>
    )
}

export default NavPages;