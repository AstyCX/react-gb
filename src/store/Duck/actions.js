//Gists
export const GET_GISTS_REQUEST = 'GISTS::GET_GISTS_REQUEST';
export const GET_GISTS_REQUEST_SAGA = 'GISTS::GET_GISTS_REQUEST_SAGA';
export const GET_GISTS_SUCCESS = 'GISTS::GET_GISTS_SUCCESS';
export const GET_GISTS_FAIL = 'GISTS::GET_GISTS_FAIL';

export const getGistsRequestSaga = (api) => ({
    type: GET_GISTS_REQUEST_SAGA,
    payload: api
});

export const getGistsRequest = (api) => ({
    type: GET_GISTS_REQUEST,
    payload: api
});

export const getGistsSuccess = (gists) => ({
    type: GET_GISTS_SUCCESS,
    payload: gists
});

export const getGistsFail = (error) => ({
    type: GET_GISTS_FAIL,
    payload: error
});


//Chats
export const CHANGE_CHAT_LIST_FB = 'CHATS::CHANGE_CHAT_LIST_FB';
export const CHANGE_CHAT_LIST_SAGA = 'CHATS::CHANGE_CHAT_LIST_SAGA';
export const DEL_CHAT_FB_SAGA = 'CHATS::DEL_CHAT_FB_SAGA';
export const ADD_CHAT_FB_SAGA = 'CHATS::ADD_CHAT_FB_SAGA';

export const addChatFBSaga = (name) => ({
    type: ADD_CHAT_FB_SAGA,
    payload: name
})

export const delChatFBSaga = (id) => ({
    type: DEL_CHAT_FB_SAGA,
    payload: id
})

export const changeChatList = (chats) => ({
    type: CHANGE_CHAT_LIST_FB,
    payload: chats
});

export const changeChatListSaga = () => ({
    type: CHANGE_CHAT_LIST_SAGA
});


//Profile
export const CHANGE_NAME = 'PROFILE::CHANGE_NAME';

export const changeName = (name) => ({
    type: CHANGE_NAME,
    payload: name
});


//Messages
export const ADD_MESSAGE_FB_SAGA = 'MESSAGES::ADD_MESSAGE_FB';
export const CHANGE_MESSAGE_LIST_FB = 'MESSAGES::CHANGE_MESSAGE_LIST_FB';

export const changeMessageList = (msg) => ({
    type: CHANGE_MESSAGE_LIST_FB,
    payload: msg
})

export const addMessageFBSaga = (id, message) => ({
    type: ADD_MESSAGE_FB_SAGA,
    payload: {id, message}
})