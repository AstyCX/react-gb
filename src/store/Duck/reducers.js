import {
    ADD_CHAT,
    DEL_CHAT,
    ADD_CHAT_MESSAGES,
    ADD_MESSAGE,
    CHANGE_NAME,
    DEL_CHAT_MESSAGES,
    GET_GISTS_SUCCESS, GET_GISTS_REQUEST, GET_GISTS_FAIL
} from "./actions";
import {AUTHORS} from "../../constants/common";

/**
 * initialState = {
 *     chatList: [
 *         {
 *             id: string,
 *             icon: React.component,
 *             title: string
 *         }
 *     ]
 * }
 */

const chatsReducer = (state = {
    chatList: []
}, action) => {
    switch (action.type) {
        case ADD_CHAT: {
            const chatList = state.chatList || [];
            const chatId = chatList.length ? +chatList.slice(-1)[0].id + 1 : 1;

            return {
                ...state,
                chatList: [
                    ...chatList,
                    {
                        id: chatId,
                        title: action.payload
                    }
                ]
            }
        }
        case DEL_CHAT: {
            const a = {
                ...state,
                chatList: []
            };
            for (let i of state.chatList) {
                if (i.id !== action.payload) {
                    a.chatList = [...a.chatList, i];
                }
            }
            return a;
        }
        default:
            return state;
    }
}

/**
 * initialState = {
 *     messageList: {
 *         id: [
 *             {
 *                 text: string,
 *                 author: string,
 *                 id: string
 *             }
 *         ]
 *     }
 * }
 */

const messagesReducer = (state = {
    messageList: {}
}, action) => {
    switch (action.type) {
        case ADD_MESSAGE: {
            const {id, message} = action.payload;
            const oldMessages = state.messageList[id] || [];

            return {
                ...state,
                messageList: {
                    ...state.messageList,
                    [id]: [
                        ...oldMessages,
                        {
                            ...message,
                            id: `${id}${oldMessages.length}`
                        }
                    ]
                }
            }
        }
        case ADD_CHAT_MESSAGES: {
            const id = Object.keys(state.messageList).length ? +Object.keys(state.messageList).slice(-1)[0]+1 : 1;

            return {
                ...state,
                messageList: {
                    ...state.messageList,
                    [id]: []
                }
            }
        }
        case DEL_CHAT_MESSAGES: {
            const a = {
                ...state,
                messageList: {}
            };
            for (let i in state.messageList) {
                if (i !== String(action.payload)) {
                    a.messageList = {...a.messageList, [i]: state.messageList[i]};
                }
            }
            return a;
        }
        default:
            return state;
    }
}

/**
 * initialState = {
 *     checked: boolean
 * }
 */

const profileReducer = (state = {
    name: AUTHORS.me
}, action) => {
    switch (action.type) {
        case CHANGE_NAME:
            return {
                ...state,
                name: action.payload
            };
        default:
            return state;
    }
}

export const STATUSES = {
    IDLE: 0,
    REQUEST: 1,
    SUCCESS: 2,
    FAILURE: 3
}

/**
 * initialState = {
 *     gists: [],
 *     request: number,
 *     error: {
 *         event: boolean,
 *         description: string
 *     }
 * }
 */

const gistsReducer = (state = {
    gists: [],
    request: STATUSES.IDLE,
    error: {
        description: '',
        event: null
    }
}, action) => {
    switch (action.type) {
        case GET_GISTS_SUCCESS:
            return {
                ...state,
                gists: action.payload,
                request: STATUSES.SUCCESS,
                error: {
                    ...state.error,
                    event: false
                }
            }
        case GET_GISTS_REQUEST:
            return {
                ...state,
                request: STATUSES.REQUEST
            }
        case GET_GISTS_FAIL:
            return {
                ...state,
                request: STATUSES.FAILURE,
                error: {
                    ...state.error,
                    description: action.payload,
                    event: true
                }
            }
        default:
            return state;
    }
}

export {
    messagesReducer,
    chatsReducer,
    profileReducer,
    gistsReducer
}