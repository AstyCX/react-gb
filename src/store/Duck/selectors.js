export const getChatList = state => state.chats.chatList;

export const getProfileName = state => state.profile.name;

export const getState = state => state;

export const getMessageList = state => state.messages.messageList;

export const getGists = state => state.gists;
