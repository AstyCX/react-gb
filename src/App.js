import './App.css';
import Routing from "./pages/containers/Routing";
import {BrowserRouter} from "react-router-dom";
import NavPages from "./components/containers/NavPages";

const App = () => {
    return (
        <BrowserRouter>
            <div className="App">
                <header className="App-header">
                    <h1>My Messenger</h1>
                    <NavPages pages={
                        {
                            Home: {
                                id: '',
                                link: '/'
                            }, Profile: {
                                id: 'profile',
                                link: '/profile'
                            }, Chats: {
                                id: 'chats',
                                link: '/chats/'
                            }
                        }
                    }/>
                </header>
                <main className='App-main'>
                    <Routing/>
                </main>
            </div>
        </BrowserRouter>
    );
}

export default App;
