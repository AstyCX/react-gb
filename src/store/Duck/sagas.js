import {ADD_MESSAGE_WITH_SAGA, addMessage} from "./actions";
import {put, delay, takeLatest} from 'redux-saga/effects'
import {AUTHORS} from "../../constants/common";

function* handleMessage(action) {
    yield put(addMessage(action.payload.id, action.payload.message));
    if (action.payload.message.author !== AUTHORS.bot) {
        const message = {text: 'Bot received the message', author: AUTHORS.bot};
        yield delay(1500);
        yield put(addMessage(action.payload.id, message));
    }
}

function* mySaga () {
    yield takeLatest(ADD_MESSAGE_WITH_SAGA, handleMessage);
}

export default mySaga;