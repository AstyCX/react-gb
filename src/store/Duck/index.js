import {createStore, combineReducers} from 'redux';
import * as reducers from './reducers';

const reducersState = combineReducers({
    profile: reducers.profileReducer,
    chats: reducers.chatsReducer,
    messages: reducers.messagesReducer
})

const store = createStore(reducersState, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());

export default store;