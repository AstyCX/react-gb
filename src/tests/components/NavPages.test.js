import {render} from "@testing-library/react";
import NavPages from "../../components/containers/NavPages";
import App from "../../App";
import {store} from "../../store/Duck";
import {Provider} from "react-redux";

describe('NavPages component test', ()=>{
    it('Returns React Component', () => {
        const rendered = render(<Provider store={store}><App pages={
            {
                Home: {
                    id: '',
                    link: '/'
                }, Profile: {
                    id: 'profile',
                    link: '/profile'
                }, Chats: {
                    id: 'chats',
                    link: '/chats'
                }, Gists: {
                    id: 'gists',
                    link: '/gists'
                }, Login: {
                    id: 'login',
                    link: '/login'
                }, Registration: {
                    id: 'registration',
                    link: '/registration'
                }
            }}><NavPages /></App></Provider>)

        expect(rendered).toMatchSnapshot();
    });
})