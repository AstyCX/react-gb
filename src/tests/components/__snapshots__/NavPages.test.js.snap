// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavPages component test Returns React Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <h1>
            My Messenger
          </h1>
          <div
            class="MuiTabs-root css-1ujnqem-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                aria-label="nav tabs example"
                class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                role="tablist"
              >
                <a
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  href="/"
                  role="tab"
                  tabindex="-1"
                >
                  Home
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
                <a
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  href="/profile"
                  role="tab"
                  tabindex="-1"
                >
                  Profile
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
                <a
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  href="/chats"
                  role="tab"
                  tabindex="-1"
                >
                  Chats
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
                <a
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  href="/gists"
                  role="tab"
                  tabindex="-1"
                >
                  Gists
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
                <a
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  href="/login"
                  role="tab"
                  tabindex="0"
                >
                  Login
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
                <a
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  href="/registration"
                  role="tab"
                  tabindex="-1"
                >
                  Registration
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
              <span
                class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </header>
        <main
          class="App-main"
        >
          <div
            class="login"
          >
            <div
              class="Toastify"
            />
            <form
              class="formLogin "
            >
              <div
                class="MuiFormControl-root MuiTextField-root logInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="mui-1"
                  id="mui-1-label"
                >
                  Your email
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-1"
                    required=""
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Your email
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root logInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="mui-2"
                  id="mui-2-label"
                >
                  Your password
                  <span
                    aria-hidden="true"
                    class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-2"
                    required=""
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Your password
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root submitLog css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Login 
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              
            </form>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <h1>
          My Messenger
        </h1>
        <div
          class="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              aria-label="nav tabs example"
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <a
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                href="/"
                role="tab"
                tabindex="-1"
              >
                Home
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                href="/profile"
                role="tab"
                tabindex="-1"
              >
                Profile
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                href="/chats"
                role="tab"
                tabindex="-1"
              >
                Chats
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                href="/gists"
                role="tab"
                tabindex="-1"
              >
                Gists
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                href="/login"
                role="tab"
                tabindex="0"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                href="/registration"
                role="tab"
                tabindex="-1"
              >
                Registration
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
            <span
              class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </header>
      <main
        class="App-main"
      >
        <div
          class="login"
        >
          <div
            class="Toastify"
          />
          <form
            class="formLogin "
          >
            <div
              class="MuiFormControl-root MuiTextField-root logInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-1"
                id="mui-1-label"
              >
                Your email
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-1"
                  required=""
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Your email
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root logInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-2"
                id="mui-2-label"
              >
                Your password
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Your password
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root submitLog css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Login 
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            
          </form>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
